### DISCUSSION TITLE
Should the <sp-textfield> elements reset in a form?

### DISCUSSION THREAD

**Discussion by peschee** [adobe/spectrum-web-components#3861] [Q&A] [answered]
I have created a simple test-case here: https://heathered-jasper-waterfall.glitch.me (https://glitch.com/edit/#!/heathered-jasper-waterfall)

The page has a simple form along with some native AND sp-* elements.

The native elements behave as expected when the reset buttons are triggered. The `sp-textfield` element, however, is not being reset.

Is this expected?

**Comment by Westbrook** [comment, marked as answer]
`<sp-textfield>` and other SWC form elements are not currently [form associated ](https://webkit.org/blog/13711/elementinternals-and-form-associated-custom-elements/), so you cannot expect the reset button to have an effect on them. If you were interested in this functionality, we'd be happy to accept and issue or a contributions, were you so inclined. @blunteshwar recently added support for the `name` attribute, which is a first step along this process, so he might be able to support any contribution you were interested in preparing.

**Reply by peschee** [reply]
Thanks @Westbrook, it is what I thought.

Were you discussing implementing the form elements like `<sp-textfield>` as "wrappers" around native elements and then slotting them in from the light DOM? I'm just interested to know what thoughts you might have gone through when implementing those elements.

**Reply by Westbrook** [reply]
At creation time, we were primarily looking to support JS powered application with these elements, so the need to do wrapped in a pre-Form Association era was not something that we saw as a requirement for this implementation. Now that our usage has expanded to even more contexts, we do have it on the backlog to move to Form Associated elements, but it is not currently on a scheduled road map.
